Tests/PackageCollectionsTests/Utility.swift

```diff
@@ -53,7 +53,7 @@ func makeMockCollections(count: Int = Int.random(in: 50 ... 100), maxPackages: I
                                                                packageName: "package-\(packageIndex)",
                                                                targets: targets,
                                                                products: products,
-                                                               toolsVersion: .currentToolsVersion,
+                                                               toolsVersion: .v5_3,
```

Comment:
interested to understand why this cant use currentToolsVersion / what the warning is about

Reply:
It's possible that this one didn't need to be changed for the 5.3 to 5.4 transition.  The specific warning came from some of the PackageToolTests, and to reduce risk and keep the version constant I searched for uses of `.currentToolsVersion` in unit tests and changed these as well.

I can set these back to `.currentToolsVersion` if it's fine that the semantics they end up with can change when the version number changes.

Reply:
The other use of it (in PackageToolTests) was a mistake, and I assumed this one was as well.  The warning was actually the one that we added for executable targets, where a package manifest that uses 5.4 and later but has a target that isn't explicitly declared as `executableTarget` gets a warning that it should adopt the new syntax.  The warning isn't shown for tools version 5.3 and earlier.  The unit test failed because it asserted that there were no warnings or errors.